src/app/modules/game/pages/game/game.component.ts

```diff
@@ -1,15 +1,24 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit } from "@angular/core";
+import { GameViewService } from "../../services/game-view.service";
+import { Observable } from 'rxjs';
 
 @Component({
-  selector: 'app-game',
-  templateUrl: './game.component.html',
-  styleUrls: ['./game.component.scss']
+  selector: "app-game",
+  templateUrl: "./game.component.html",
+  styleUrls: ["./game.component.scss"],
 })
 export class GameComponent implements OnInit {
+  currentView: string;
 
-  constructor() { }
+  constructor(private gameViewService: GameViewService) {}
 
   ngOnInit(): void {
+    this.initGameView();
   }
 
+  initGameView(): void {
+    this.gameViewService.currentView$.subscribe((view: string) => {
```

Comment:
You need to unsubscribe from observable when component is destroyed. Use "takeUnitl" inside pipe

Reply:
I think I only call subscribe from the service, without storing actual subscriptions inside the service. Or am I thinking wrong here?